Allow loop device for installation target

dbnicholson opened the pull request:
The requirement to have an actual non-removable disk on hand for testing is pretty onerous. This allows using a loop device for the installation target and ensures that it's not the same if the source is also a loop device.

I feel like you could probably do some automated testing with this presuming that the test was run from a fairly privileged process.

dbnicholson committed:
disktarget: Allow using loop device for install target

This greatly simplifies testing by not requiring an actual drive for the
target. A large enough file can be associated with a loop device and it
will be considered for installation.

dbnicholson committed:
disktarget: Ensure target loop doesn't overwrite source loop

When both the source is a loop device, don't present it as a valid
target device now that loops are allowed there.

wjt commented on gnome-image-installer/pages/disktarget/gis-disktarget-page.c:
```diff
@@ -287,51 +287,79 @@ gis_disktarget_page_populate_model(GisPage *page, UDisksClient *client)
+
+          targetname = g_strdup_printf("%s %s",
+                                       udisks_block_get_device (block),
+                                       udisks_loop_get_backing_file (loop));
```
You can use `backing_file` here

wjt reviewed (approved):
LGTM, besides one superficial change.

All this logic (including the existing logic) feels a bit messy, but this seems to be the fault of the underlying APIs (UDisks and below).

dbnicholson committed:
README: Document a process for testing with loop devices

This should cover a whole end to end procedure for running the installer
without using actual disks.

dbnicholson commented:
I made that change and added some documentation on using loop devices for testing.
